src/compiler/crystal/codegen/target_machine.cr

```diff
@@ -19,7 +22,7 @@ module Crystal
           features += "+vfp2"
         end
       else
-        raise "Unsupported arch for target triple: #{target_triple}"
+        raise TargetMachine::Error.new("Unsupported architecture for target triple: #{target_triple} (only x86, x86_64, and ARM are supported)")
```

Comment:
I'm not sure if the supported architectures should be hardcoded in the error message. Might be missed when (and if) say PowerPC would be supported someday.
Maybe just provide a link to https://github.com/crystal-lang/crystal/wiki/Platform-Support (the wiki page is still a draft though).